pkg/core/function.go

```diff
@@ -581,12 +553,46 @@ func (c *client) UpdateFunction(buildpackBuilder Builder, options UpdateFunction
 	return nil
 }
 
-func buildLocally(builder Builder, appDir string, buildImage string, runImage string, repoName string) error {
-	if buildImage == "" {
+func doBuildLocally(builder Builder, image string, options BuildOptions) error {
+	if err := writeRiffToml(options); err != nil {
+		return err
+	}
+	defer func() { _ = deleteRiffToml(options) }()
+	if options.BuildpackImage == "" {
 		return fmt.Errorf("unable to build function locally: buildpack image not specified")
 	}
-	if runImage == "" {
+	if options.RunImage == "" {
 		return fmt.Errorf("unable to build function locally: run image not specified")
 	}
-	return builder.Build(appDir, buildImage, runImage, repoName)
+	return builder.Build(options.LocalPath, options.BuildpackImage, options.RunImage, image)
+}
+
+func writeRiffToml(options BuildOptions) error {
+	t := struct {
+		Override string `toml:"override"`
+		Handler  string `toml:"handler"`
+		Artifact string `toml:"artifact"`
+	}{
+		Override: options.Invoker,
```

Comment:
Again, it would make sense to rename `Override` to `Invoker`.

Reply:
This is part of the riff.toml (hidden) contract. Could tackle in a different PR, but would require some coordination